validator/src/main.rs

```diff
@@ -212,7 +212,9 @@ fn get_rpc_node(
         );
 
         if rpc_peers_blacklisted == rpc_peers_total {
```

Comment:
Note that: When no nodes is found, this evaluates to `if 0 == 0 {`... So, it never resets `blacklist_timeout`.